sample/kotlin/src/main/kotlin/com/datadog/android/sample/widget/WidgetIntentService.kt

```diff
@@ -0,0 +1,100 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sample.widget
+
+import android.app.IntentService
+import android.appwidget.AppWidgetManager
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.view.View
+import android.widget.RemoteViews
+import com.datadog.android.rum.GlobalRum
+import com.datadog.android.rum.RumActionType
+import com.datadog.android.sample.R
+import com.datadog.android.sample.SampleApplication
+import okhttp3.Request
+import timber.log.Timber
+
+class WidgetIntentService : IntentService("WidgetIntentService") {
+
+    override fun onHandleIntent(intent: Intent?) {
+        when (intent?.action) {
+            LOAD_RANDOM_RESOURCE_ACTION -> {
+                handleActionWithRumContext(intent) {
+                    updateUIStatus(applicationContext, true)
+                    val okHttpClient = SampleApplication.getOkHttpClient(applicationContext)
+                    val builder = Request.Builder()
+                        .get()
+                        .url("https://www.datadoghq.com/")
+
+                    val request = builder.build()
+                    try {
+                        val response = okHttpClient.newCall(request).execute()
+                        val body = response.body()
+                        if (body != null) {
+                            val content: String = body.string()
+                            // Necessary to consume the response
+                            Timber.d("Response: $content")
+                        }
+                    } catch (e: Exception) {
+                        Timber.e("Error: ${e.message}")
+                    }
+                    updateUIStatus(applicationContext, false)
+                }
+            }
+            else -> {
+            }
+        }
+    }
+
+    private inline fun handleActionWithRumContext(
```

Comment:
Why do you need to use lambda when it's not necessary ? You could as easily make a method to start the request and call it in there :/

Reply:
in case you have multiple actions ....you don't want to add this boiler code all the time (the one that starts/stop the view and adds the action event). Instead you can use this method to actually decorate your lambda with the rum events.